InvasivesBC/ViewControllers/BaseViewController.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  BaseViewController.swift
+//  ipad
+//
+//  Created by Amir Shayegh on 2019-10-29.
+//  Copyright © 2019 Amir Shayegh. All rights reserved.
+//
+
+import UIKit
+import DatePicker
+
+protocol InputDelegate: class {
+    func showDropdownDelegate(items: [DropdownModel], on view: UIView, callback: @escaping (_ selection: DropdownModel?) -> Void)
+    func showDatepickerDelegate(on view: UIView, initialDate: Date?, minDate: Date?, maxDate: Date?, callback: @escaping (_ date: Date?) -> Void)
+    func showTimePickerDelegate(on view: UIView, initialTime: Time?, completion: @escaping (_ time: Time?) -> Void)
+    func showOptionsDelegate(options: [OptionType], on button: UIButton, completion: @escaping (_ option: OptionType) -> Void)
+}
+
+class BaseViewController: UIViewController, Theme ,InputDelegate {
+    
+    // MARK: Constants
+    let visibleAlpha: CGFloat = 1
+    let invisibleAlpha: CGFloat = 0
+    
+    // MARK: Variables
+    var currentPopOver: UIViewController?
+    
+    // MARK: ViewController Functions
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    // MARK: Event handlers
+    func whenLandscape() {}
+    func whenPortrait() {}
+    func orientationChanged() {}
+    
+    // MARK: Utilities
+    public func dismissKeyboard() {
+        self.view.endEditing(true)
+    }
+    
+    public func setNavigationBar(hidden: Bool, style: UIBarStyle) {
+        if let navigationController = self.navigationController {
+            navigationController.navigationBar.isHidden = hidden
+            navigationController.navigationBar.barStyle = style
+        }
+    }
+    
+    // MARK: Popover
+    private func showPopOver(on button: UIButton, popOverVC: UIViewController, height: Double, width: Double, arrowColor: UIColor? = nil, direction: UIPopoverArrowDirection? = .any) {
```

Comment:
UIButton is a subclass of UIControl which is a child of UIView. So, this overloaded version is redundant.